crates/bevy_reflect/src/impls/std.rs

```diff
@@ -707,14 +706,23 @@ impl<T: FromReflect> Reflect for Option<T> {
                     "Some" => {
                         let field = value
                             .field_at(0)
-                            .expect("Field at index 0 should exist")
-                            .clone_value();
-                        let field = field.take::<T>().unwrap_or_else(|_| {
-                            panic!(
-                                "Field at index 0 should be of type {}",
-                                std::any::type_name::<T>()
-                            )
-                        });
+                            .unwrap_or_else(|| {
+                                panic!(
+                                    "Field in `Some` variant of {} should exist",
+                                    std::any::type_name::<Option<T>>()
+                                )
+                            })
+                            .clone_value()
+                            .take::<T>()
+                            .unwrap_or_else(|value| {
```

Comment:
Seems reasonable for the current impl. Stuff like this makes me want to revisit mixing "dynamic type" reflection and "actual type" reflection. But thats a conversation for another day :)

Reply:
Sounds like a good time to revisit https://github.com/bevyengine/rfcs/pull/56 😉